qa-tests-backend/scripts/run-compatibility.sh

```diff
@@ -119,4 +124,34 @@ shorten_tag() {
     fi
 }
 
+verify_scannerV2_deployed() {
+    local namespace=${1:-stackrox}
+    info "Waiting for Scanner V2 deployment to appear in namespace ${namespace}..."
+    wait_for_object_to_appear "$namespace" deploy/scanner-db 600
+    wait_for_object_to_appear "$namespace" deploy/scanner 300
+    info "** Scanner V2 is deployed in namespace ${namespace}"
+}
+
+verify_scannerV4_deployed() {
```

Comment:
As far as I've checked it's only `CertExpiryTest`  that requires scanner to be available, and it does not run any actual image scans.
https://github.com/stackrox/stackrox/blob/bb42b281e1b728abc30e4cc75318160396f57a85/qa-tests-backend/src/test/groovy/CertExpiryTest.groovy#L13-L42

It then makes sure that the scanner TLS cert and the scanner cert expiry stored in central match, which is why it falls under compatibility. It looks like you're right and scanner v2 might be enough.

> Maybe we can reach out to the scanner team and ask if they have some kind of pre-initialized test-DB for running in tests.

I find that unlikely, because I've taken the functions I to check if scanner is deployed from the `run-scanner-v4.bats` tests.
https://github.com/stackrox/stackrox/blob/77048a6a739e0ed649926899d4ef9432506c1ae3/tests/e2e/run-scanner-v4.bats

> Scanner v4 takes 20-45 minutes to initially populate its DB.

Perhaps it would make more sense to add these scanner readiness checks to the CertExpiryTest itself(?) The problem with that is that it's a `.groovy` file and I would first have to learn the syntax for that.

Or there is the option of removing CertExpiryTest from compatibility testing if it adds too much overhead, as suggested by Gavin in https://issues.redhat.com/browse/ROX-24959

Reply:
I see that getting the scanner cert from secret is easy, but that is then later compared against an entry in Central DB. How that entry is created? If deployment of scanner creates it, then we need to deploy scanner and make sure that is is registered in Central. This needs to be checked

Reply:
I've checked and central does indeed need a connection to scanner to get the cert expiry date. So I've added a kubectl wait for the scanner pod to be ready.